src/sinol_make/helpers/compile.py

```diff
@@ -0,0 +1,44 @@
+import sinol_make.helpers.compiler as compiler
+from sinol_make.interfaces.Errors import CompilationError
+import os, subprocess, sys
+
+def compile(program, output, compilers = None, compile_log = None):
+	"""
+	Compile a program
+	compilers - A dictionary of compilers to use. If not set, the default compilers will be used
+	"""
+	ext = os.path.splitext(program)[1]
+	arguments = []
+	if ext == '.cpp':
+		arguments = [compilers['cpp_compiler_path'] or compiler.get_cpp_compiler_path(), program, '-o', output] + \
+						'-O3 -lm -Werror -Wall -Wextra -Wshadow -Wconversion -Wno-unused-result -Wfloat-equal'.split(' ')
```

Comment:
can we make this as a variable/argument and then just pass `C_FLAGS`?

Reply:
Maybe we should add flags to config.yml and if none are set then the code should use less strict flags then those that are now used?

Reply:
Or by default the flags should be the same as those that oioioi uses

Reply:
It would indeed be beneficial to be able to pass custom flags, e.g. `-fsanitize=address,undefined` or `-D_GLIBCXX_DEBUG`, or temporarily disable one of the warning flags.

But then, the calculated times won't necessarily be the same as the ones on sio2 and it might also trigger the `sinol_expected_scores` mechanism. If we want to have the ability to specify flags, it should give a warning that the times might not be the same and ignore the `sinol_expected_scores`. Probably later we would want to have a `sinol-make verify` that runs each command and checks whether they work as expected (e.g. the input files haven't changed their hashes, they satisfy the `inwer`, the output files are correct, the `sinol_expected_scores` is set as expected, etc) and in this command we wouldn't want the user to specify the custom flags.

To do it perfectly we might need to put into it more thought and the implementation won't be trivial. I think, at this point, it's better to create an Issue for later, focus on merging this PR and adding non-necessary functionality after writing basic tests.

Reply:
I created the issue #4, so I think that we can merge this PR